tests/unit/test_charm.py

```diff
@@ -345,46 +347,52 @@ def test_db_relation(self):
+    @patch.object(Container, "exec")
+    def test_add_admin_user(self, mock_exec):
         """
         arrange: an email and a password
         act: when the _on_add_admin_user_action mtehod is executed
-        assert: the underlying rake command to add the user is executed with the appropriate parameters.
+        assert: the underlying rake command to add the user is executed
+        with the appropriate parameters.
         """
+        mock_exec.return_value = MagicMock(wait_output=MagicMock(return_value=("", None)))
         self.harness.disable_hooks()
         self.add_database_relations()
 
         charm: DiscourseCharm = typing.cast(DiscourseCharm, self.harness.charm)
-        with patch.object(Container, "exec", return_value=MockExecProcess()) as exec_mock:
-            self.harness.container_pebble_ready("discourse")
-
-            email = "sample@email.com"
-            password = "somepassword"  # nosec
-            event = MagicMock(spec=ActionEvent)
-            event.params = {
-                "email": email,
-                "password": password,
-            }
-            charm._on_add_admin_user_action(event)
+        self.harness.container_pebble_ready("discourse")
 
-        exec_mock.assert_any_call(
+        email = "sample@email.com"
+        password = "somepassword"  # nosec
+        event = MagicMock(spec=ActionEvent)
+        event.params = {
+            "email": email,
+            "password": password,
+        }
+        charm._on_add_admin_user_action(event)  # pylint: disable=protected-access
+
+        mock_exec.assert_any_call(
             [
                 "bash",
                 "-c",
                 f"./bin/bundle exec rake admin:create <<< $'{email}\n{password}\n{password}\nY'",
             ],
             user="discourse",
             working_dir=DISCOURSE_PATH,
-            environment=charm._create_discourse_environment_settings(),
+            environment=charm._create_discourse_environment_settings(),  # pylint: disable=protected-access
         )
 
     def add_postgres_relation(self):
         "Add postgresql relation and relation data to the charm."
-        self.harness.charm._stored.db_name = "discourse-k8s"
-        self.harness.charm._stored.db_user = "someuser"
-        self.harness.charm._stored.db_password = "somepasswd"  # nosec
-        self.harness.charm._stored.db_host = "dbhost"
-        self.db_relation_id = self.harness.add_relation("db", "postgresql")
+        self.harness.charm._stored.db_name = "discourse-k8s"  # pylint: disable=protected-access
+        self.harness.charm._stored.db_user = "someuser"  # pylint: disable=protected-access
+        self.harness.charm._stored.db_password = (  # pylint: disable=protected-access
+            "somepasswd"  # nosec
+        )
+        self.harness.charm._stored.db_host = "dbhost"  # pylint: disable=protected-access
+        self.db_relation_id = (  # pylint: disable=attribute-defined-outside-init
```

Comment:
Should we note why this is reasonable to do?

Reply:
Good point, noted!